main.c

```diff
@@ -0,0 +1,91 @@
+#include <string.h>
+#include "main.h"
+#include "_cgo_export.h"
+
+result *save_to_png(size_t max_store, int page_number, int width, float scale, const unsigned char *payload, size_t payload_length) {
+	if (max_store == 0) {
+		max_store = FZ_STORE_DEFAULT;
+	}
+
+	fz_context *ctx = fz_new_context(NULL, NULL, max_store);
+	if (!ctx) {
+		errorHandler("fail to create mupdf context");
+		return NULL;
+	}
+
+	result *r = malloc(sizeof(result));
+	r->context = ctx;
+	r->buffer = NULL;
+
+	int exit = 0;
+	fz_stream *stream = NULL;
+	fz_document *doc = NULL;
+	fz_page *page = NULL;
+	fz_display_list *list = NULL;
+	fz_device *device = NULL;
+	fz_pixmap *pixmap = NULL;
+	fz_device *draw_device = NULL;
+	fz_buffer *buffer = NULL;
+	fz_try(ctx) {
+		fz_register_document_handlers(ctx);
+		stream = fz_open_memory(ctx, payload, payload_length);
+		doc = fz_open_document_with_stream(ctx, "document.pdf", stream);
```

Comment:
It requires something there to work:
```
/**
	Open a document using the specified stream object rather than
	opening a file on disk.

	magic: a string used to detect document type; either a file name
	or mime-type.
*/
```
This is from when I was using the filesystem but still works like this.